common/content/finder.js

```diff
@@ -220,23 +223,19 @@ var Finder = Module("finder", {
             this.find(pattern);
         }
 
-        this._lastSearchPattern = pattern;
-
         // TODO: move to find() when reverse incremental searching is kludged in
         // need to find again for reverse searching
         if (this._backwards)
             this.findAgain(true);
 
         if (options["hlsearch"])
-            this.highlight(findbar._findField.value);
+            this.highlight();
     },
 
     /**
      * Highlights all occurances of <b>str</b> in the buffer.
-     *
-     * @param {string} str The string to highlight.
      */
-    highlight: function (str) {
+    highlight: function () {
```

Comment:
They can if they go through finder directly like that (I'm not sure how much trouble doing that would cause in terms of keeping the findbar and its results in sync though, never really tried it), but since this method is now going through the findbar, that argument is not necessary in this case.